### PR TITLE
Attachment: create any and return more info

### PR DESCRIPTION
**Pull request by TimPetricola**
# Why

While working on Mind oncall process I can't setup a Zap to post `@care` comments from Linear to Intercom, as the Linear project can be linked to a single Intercom workspace.

I believe it's still doable by creating the Linear attachment manually (not relying on `attachmentLinkIntercom`) with custom metadata, and then retrieving it

# How

- add new `attachmentCreate` action, which can also take a subtitle and metadata
- in `issueAttachments`, return title, subtitle and URL (not just metadata)

# Test

I couldn't manage to set up OAuth on a test app, so I didn't fully test the changes. What I checked:
- `yarn zapier-validate` is fine
- GraphQL query and mutation are correct
- `dict` fields (for metadata) return a JSON

### CODE REVIEW

**Comment by agerlic** [comment]
@TimPetricola 

## To create the attachment
To create the attachment you can use https://github.com/alan-eu/linear-zapier/blob/master/src/creates/createAttachmentLinkURL.ts it's what Internation uses to create their Linear cards, eg: https://linear.app/alan-eu/issue/ES-448/🇪🇸-eng

## To retrieve the attachment
I agree we need to change the `issueAttachments` response. I'm wondering if adding the URL is not enough.

**Review by agerlic — CHANGES_REQUESTED** [review]
@TimPetricola 

## To create the attachment
To create the attachment you can use https://github.com/alan-eu/linear-zapier/blob/master/src/creates/createAttachmentLinkURL.ts it's what Internation uses to create their Linear cards, eg: https://linear.app/alan-eu/issue/ES-448/🇪🇸-eng

## To retrieve the attachment
I agree we need to change the `issueAttachments` response. I'm wondering if adding the URL is not enough.

**Comment by TimPetricola** [comment]
@agerlic I initially planned on adding the `conversationId` as a metadata (that's the reason for not relying on `createAttachmentLinkUrl`). But actually I believe you're right, I can just use retrieve the URL and extract the conversation id from there.

I've updated the PR, thanks for simplifying :)

File: src/searches/issueAttachments.ts [hunk 1]
```diff
@@ -28,72 +27,73 @@ interface issueAttachmentsRequestResponse {
...
+              id,
+              title,
+              subtitle,
               metadata
             }
           }
         }
       }`;
 
-      z.console.log(JSON.stringify(query));
```
Comment (agerlic) [inline src/searches/issueAttachments.ts]: NIT: the diff is not super easy to read as all indentation has changed

File: src/searches/issueAttachments.ts [hunk 2]
```diff
@@ -28,72 +27,73 @@ interface issueAttachmentsRequestResponse {
           }
           attachments {
             nodes {
+              id,
```
Comment (agerlic) [inline src/searches/issueAttachments.ts:31]: the URL is missing is it expected ?

**Review by agerlic — CHANGES_REQUESTED** [review]
Both can work you're right.
Let's try with that one first and if it's too complex we can add the new createAttachment endpoint.

File: src/searches/issueAttachments.ts (hunk 2 above)
Comment (TimPetricola) [reply src/searches/issueAttachments.ts:31]: Definitely not, went too fast 🤦 thanks for the catch

File: src/searches/issueAttachments.ts (hunk 1 above)
Comment (TimPetricola) [reply src/searches/issueAttachments.ts]: Prettier ran automatically, but you can hide whitespaces in the review:
Screenshot 2022-03-23 at 13 22 13

Comment (TimPetricola) [reply src/searches/issueAttachments.ts]: For the sake of consistency, I've reset formatting to what we have on `master`

**Review by agerlic — APPROVED** [review]
🚢

**Pull request merged**